src/nvim/highlight.c

```diff
@@ -1094,6 +1141,10 @@ HlAttrs dict2hlattrs(Dict(highlight) *dict, bool use_rgb, int *link_id, Error *e
     hlattrs.cterm_ae_attr = mask;
   }
 
+  if (HAS_KEY_X(dict, url)) {
```

Comment:
is this actually used anywhere? in the API layer the PR guards against this, but this breaks the memoization allocation pattern used above and for future usages it is better to leave a missing implementation to fill in, than a broken one.

Reply:
It's used when called from `ui_client_dict2hlattrs`. Commenting this out makes the new test in `tui_spec` fail.

We could move these lines from `dict2hlattrs` into `ui_client_dict2hlattrs` so that it only affects the TUI.

Reply:
415df202ea0cebe6b702b39022bdf191feba79c6